app/src/main/java/id/rizmaulana/covid19/data/repository/AppRepository.kt

```diff
@@ -97,23 +97,47 @@ open class AppRepository constructor(
         return Observable.concatArrayEager(localObservable, remoteObservable)
     }
 
-    override fun confirmed() = api.confirmed()
-        .flatMap {
-            setCacheConfirmed(it)
-            Observable.just(it)
-        }
+    override fun confirmed(): Observable<List<CovidDetail>> {
+        val cacheConfirmed = getCacheConfirmed()
+        val localObservable = if(cacheConfirmed != null) Observable.just(cacheConfirmed)
+        else Observable.empty()
 
-    override fun deaths() = api.deaths()
-        .flatMap {
-            setCacheDeath(it)
-            Observable.just(it)
-        }
+        val remoteObservable = api.confirmed()
+            .flatMap {
+                setCacheConfirmed(it)
+                Observable.just(it)
+            }
 
-    override fun recovered() = api.recovered()
-        .flatMap {
-            setCacheRecovered(it)
-            Observable.just(it)
-        }
+        return Observable.concatArrayEager(localObservable, remoteObservable)
+    }
+
+    override fun deaths(): Observable<List<CovidDetail>> {
+        val cacheDeath = getCacheDeath()
+        val localObservable = if(cacheDeath != null) Observable.just(cacheDeath)
+        else Observable.empty()
+
+        val remoteObservable = api.deaths()
+            .flatMap {
+                setCacheDeath(it)
+                Observable.just(it)
+            }
+
+        return Observable.concatArrayEager(localObservable, remoteObservable)
+    }
+
+    override fun recovered(): Observable<List<CovidDetail>>{
+        val cacheRecovered = getCacheRecovered()
+        val localObservable = if(cacheRecovered != null) Observable.just(cacheRecovered)
+        else Observable.empty()
+
+        val remoteObservable = api.recovered()
+            .flatMap {
+                setCacheRecovered(it)
+                Observable.just(it)
+            }
+
+        return Observable.concatArrayEager(localObservable, remoteObservable)
```

Comment:
Can you put documentation why you are using `concatArrayEager` operation?

Reply:
Okay! will do